setuptools/command/sdist.py

```diff
@@ -126,11 +126,20 @@ def _add_defaults_python(self):
         if self.distribution.has_pure_modules():
             build_py = self.get_finalized_command('build_py')
             self.filelist.extend(build_py.get_source_files())
+
             # This functionality is incompatible with include_package_data, and
             # will in fact create an infinite recursion if include_package_data
             # is True.  Use of include_package_data will imply that
             # distutils-style automatic handling of package_data is disabled
-            if not self.distribution.include_package_data:
+
+            if self.distribution.include_package_data:
+                self.warn(
```

Comment:
It looks to me like this warning may be issued unconditionally, even if package_data was not indicated. That's going to cause a lot of spurious warnings for use-cases that don't use package_data but use include_package_data. I'll test to confirm.

Reply:
Confirmed. Making this change to the tests, the tests still pass:

```diff
diff --git a/setuptools/tests/test_manifest.py b/setuptools/tests/test_manifest.py
index 25e65ce5..a1d90aad 100644
--- a/setuptools/tests/test_manifest.py
+++ b/setuptools/tests/test_manifest.py
@@ -611,7 +611,6 @@ class TestFileListTest(TempDirTestCase):
         self.make_files([ml('app/__init__.py'), ml('app/e.py')])
 
         setup_attrs = {
-            "package_data": {"some": ["data"]},
             "include_package_data": True,
             'packages': ['app'],
         }

```